sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -565,7 +582,43 @@ protected void beforeCreating() {
 
         NicIpConfigurationImpl.ensureConfigurations(this.nicIPConfigurations.values());
 
+        if (Objects.nonNull(this.specifiedIpConfigNames)) {
+            NicIpConfigurationImpl.deleteOptionConfigurations(this.nicIPConfigurations.values(), this.specifiedIpConfigNames);
+            this.isInUpdateIpConfigMode = false;
+        }
+
         // Reset and update IP configs
         this.innerModel().withIpConfigurations(innersFromWrappers(this.nicIPConfigurations.values()));
     }
+
+    @Override
+    public NetworkInterfaceImpl withPublicIPAddressDeleteOptions(DeleteOptions deleteOptions, String... ipConfigNames) {
+        Set<String> ipConfigNameSet = Arrays.stream(ipConfigNames).map(ipConfigName -> ipConfigName.toLowerCase(Locale.ROOT)).collect(Collectors.toSet());
+        if (!isInCreateMode() && !isInUpdateIpConfigMode) {
+            this.nicIPConfigurations.values()
+                .stream().filter(nicIpConfiguration ->
+                    (ipConfigNameSet.isEmpty() || ipConfigNameSet.contains(nicIpConfiguration.name().toLowerCase(Locale.ROOT)))
```

Comment:
There's a case when user accidentally passes in an emtpy array for `ipConfigNames`. We probably don't want to set all existing public IPs.